shiny/ui/_theme_brand.py

```diff
@@ -278,35 +193,38 @@ def __init__(
         )
 
         self.brand = brand
+        self.add_sass_layer_file(
+            path_pkg_www("..", "py-shiny", "brand", "_brand-yml.scss")
+        )
 
         # Prep Sass and CSS Variables -------------------------------------------------
-        sass_vars_theme_colors, sass_vars_brand_colors, css_vars_brand = (
-            ThemeBrand._prepare_color_vars(brand)
-        )
-        sass_vars_typography = ThemeBrand._prepare_typography_vars(brand)
-
-        # Theme -----------------------------------------------------------------------
-        # Defaults are added in reverse order, so each chunk appears above the next
-        # layer of defaults. The intended order in the final output is:
-        # 1. "Brand" Color palette
-        # 2. "Brand" Bootstrap Sass vars
-        # 3. "Brand" theme colors
-        # 4. "Brand" typography
-        # 5. Gray scale variables from "Brand" fg/bg or black/white
-        # 6. Fallback vars needed by additional "Brand" rules
-
-        self.add_defaults("", "// *---- brand: end of defaults ----* //", "")
-        self._add_sass_ensure_variables()
-        self._add_sass_brand_grays()
-        self._add_defaults_hdr("typography", **sass_vars_typography)
-        self._add_defaults_hdr("theme colors", **sass_vars_theme_colors)
+        (
+            brand_color_palette_defaults,
+            brand_color_defaults,
+            brand_color_palette_rules,
+        ) = ThemeBrand._prepare_color_vars(brand)
+
+        brand_typography_defaults = ThemeBrand._prepare_typography_vars(brand)
+
+        # Defaults ----
+        # Final order is reverse-insertion:
+        # * brand.color.palette
+        # * brand.defaults (Brand-defined Bootstrap defaults)
```

Comment:
I would have expected the ordering here to be swapped (i.e., Bootstrap defaults first since they're the most specific). What's the rationale for the palette coming first?

Reply:
The goal is to be able to say something like `$navbar-bg: $brand-blue`, which requires this order:

```scss
$brand-blue: "#2233ff";

$navbar-bg: $brand-blue;
```

Reply:
Yup, came from here: https://github.com/quarto-dev/quarto-cli/blob/bf550cf7049f19d486b7410cb746b70367d0c359/src/core/sass/brand.ts#L31-L34

Happy to leave this for a follow-up PR.

Reply:
Ah, ok, that feels worth it. A mention of that capability in the Sass would be great.

Reply:
idk this feels a lot like the kind of comment that's explained by the code. `brand.color.palette` is only the brand's named colors and it's Sass variables come before `brand.defaults` because otherwise they wouldn't be usable. If it were the other way around, I think it'd need to be commented.